### PR TITLE
smoke test of room by Brayan

### PR DESCRIPTION
**Pull request by BrayanRosas**

### CODE REVIEW

**Comment by AlejandroJeanCarlo** [comment]
Please don´t forget to use the Directory tests/Smoke to Smoke Tcs

**Comment by AlejandroJeanCarlo** [comment]
remember always pull the code from the main repository before push your code

File: lib/mongodb.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/**
+* mongodb.js by Brayan Rosas
+*/
+var config=require(GLOBAL.initialDirectory+'/config/config');
+var MongoClient = require('mongodb').MongoClient;
+var ObjectId = require('mongodb').ObjectID;
+var url=config.urlmongodb;
+	/**
```
Comment (AlejandroJeanCarlo) [inline lib/mongodb.js]: fix please the tab

File: lib/mongodb.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	 * @findDocument(col,json,callback)
+	 * col : is the collection of where want to search the document
+	 * json : is a json , file where should be specified the property 
+	 * for wich want to search a doument and collback : is a asynchronous function
+	 * note: if you send a empty json , the function return all documents.
+	 * @return : It function return a callback with the document fouded in jason format.
+	 */
+var findDocument = function(col,json,callback) {
+
+		MongoClient.connect(url, function(err, db) {
```
Comment (AlejandroJeanCarlo) [inline lib/mongodb.js]: tab please

File: lib/roomManagerAPI.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 //RoomManagerAPI.js
-var config=require(GLOBAL.initialDirectory+'/config/config');
+
```
Comment (AlejandroJeanCarlo) [inline lib/roomManagerAPI.js]: could you please delete this line

File: tests/Smoke/rooms.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
+var init = require('../init');
+var expect = require('chai').expect;
+
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: delete

File: tests/Smoke/rooms.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+var tokenAPI=require(GLOBAL.initialDirectory+'/lib/tokenAPI');
+var config =require(GLOBAL.initialDirectory+'/config/config.json');
+var endPoints=require(GLOBAL.initialDirectory+'/config/endPoints');
+
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: same

File: tests/Smoke/rooms.js [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+	});	
+
+/**
+ * Smoke Test to the service room with the method put for modify the
+ * display name of the room 
+ */
+	it('PUT /room/{roomId}, Verify the status 200',function(done)
+	{			
+		endPoint=config.url+endPoints.rooms+'/'+room._id;
+		json={"customDisplayName": "customDisplayName Modified"};
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: this Json should be in a config file

File: tests/Smoke/rooms.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+		roomManagerAPI.
+			put(token,endPoint,json,function(err,res){
+				expect(res.status).to.equal(200);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: replace 200 with config.httpStatus.Ok

File: tests/Smoke/rooms.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+		tokenAPI
+			.getToken(function(err,res){
+				token = res.body.token;
+				endPoint=config.url+endPoints.rooms;
+				roomManagerAPI
+					.get(endPoint,function(err,resp){
+						randomRoom=util.generateRandom(1,resp.body.length);
+						room=resp.body[randomRoom];
+						
+						endPoint=config.url+endPoints.resources
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: this line should be declared at the begin of the code

File: tests/Smoke/rooms.js [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+						json=util.getRandomResourcesJson(resourceConfig.resourceNameSize);
+							roomManagerAPI.post(token,endPoint,json,function(err,resourceRes){
+								resource=resourceRes;
+									var endPoint2=config.url+endPoints.rooms+'/'+room._id+'/resources';
+									var json2={"resourceId": resource.body._id,"quantity": 1 };
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: this Json should be obtained from a config file

File: tests/Smoke/rooms.js [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+			});	
+	});
+
+/**
+ * Smoke Test to the service room with the method get for getting a resource for
+ * a specific room .
+ */
+	it('GET /rooms/{roomId}/resources,Verify the status 200',function(done){	
+
+		 			endPoint=config.url+endPoints.rooms+'/'+room._id+endPoints.resources;
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: fix the tab here please

File: tests/Smoke/rooms.js [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+		 			roomManagerAPI.get(endPoint,function(err,res){
+		 				expect(res.status).to.equal(200);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: same (200)

File: tests/Smoke/rooms.js [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+		 			});			  				  			 						
+	});	
+
+/**
+ * Smoke Test to the service room with the method POST 
+ * for associates the room with an existent resource.
+ */
+	it('POST rooms/{:roomId}/resources, Verify the status 200',function(done){
+			
+					expect(resourceAsoc.status).to.equal(200);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: tab

File: tests/Smoke/rooms.js [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+				})
+	});	
+
+ /**
+ * Smoke Test to the service room with the method PUT
+ * for Updating a specific resource from a specific room
+ */			
+	it('PUT /rooms/{:roomId}/resources/{:roomResourceId}, Verify the status 200',function(done){
+				
+				endPoint=config.url+endPoints.rooms+'/'+room._id+endPoints.resources+'/'+
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: tab

File: tests/Smoke/rooms.js [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+ /**
+ * Smoke Test to the service room with the method DEL
+ * for Removing a specific resource from a specific room
+ */	
+	it('DELETE /rooms/{:roomId}/resources/{:roomResourceId}, Verify the status 200 ',function(done){
+						
+			endPoint=config.url+endPoints.rooms+'/'+room._id+endPoints.resources+'/'+
+					 resourceAsoc.body.resources[0]._id;
+					 roomManagerAPI.del(token,endPoint,function(err,resp){
+					 	expect(resp.status).to.equal(200);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: same

**Comment by AlejandroJeanCarlo** [comment]
please review the new observations

File: lib/roomManagerAPI.js (hunk 3 above)
Comment (BrayanRosas) [inline lib/roomManagerAPI.js]: done

File: tests/Smoke/rooms.js (hunk 4 above)
Comment (BrayanRosas) [inline tests/Smoke/rooms.js]: done

File: tests/Smoke/rooms.js (hunk 5 above)
Comment (BrayanRosas) [inline tests/Smoke/rooms.js]: done

File: config/endPoints.json [hunk 15]
```diff
@@ -4,7 +4,11 @@
 	"resources" :"/resources",
 	"login" : "/login",
 	"resourceId":"/resources/{:id}",
+	"rooms": "/rooms",
+  	"roomsId": "\/rooms\/{roomId}",
```
Comment (AlejandroJeanCarlo) [inline config/endPoints.json]: please use only /

File: config/endPoints.json [hunk 16]
```diff
@@ -4,7 +4,11 @@
...
+  	"roomsIdResources":"\/rooms\/{roomId}\/resources",
+  	"roomsIdResourcesId":"\/rooms\/{:roomId}\/resources\/{roomResourceId}",
```
Comment (AlejandroJeanCarlo) [inline config/endPoints.json]: same

File: tests/Smoke/rooms.js [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+ * Smoke Test to the service room with the method get for roomId getting 
+ * randondly.
+ */
+	it('Get /rooms/{roomId}, Verify the status 200 ',function(done){
+		
+		endPoint=config.url+endPoints.rooms+'/'+room._id;
+
+		roomManagerAPI.
+			get(endPoint,function(err,res){
+				expect(res.status).to.equal(200);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: same

File: tests/Smoke/rooms.js [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+				done();
+			});
+
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: delete

File: tests/Smoke/rooms.js [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+describe('Smoke Testing for Room Resources routes ', function() {
+	this.timeout(config.timeOut);
+
+	/**
+	 * Pre condition to execute the set Test Cases.
+	 * @getToken(rollback)
+	 * Obtain a token to an user account setting in the config.json file,
+	 * Get a room randomly and create a resource
+	 */
+	before('Before Set',function (done) {
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/rooms.js]: could you please re fact this method

**Comment by BrayanRosas** [comment]
Jean Carlo, please review mi code , all modifications that you asked me , are done , please review them , and merge mi code

File: tests/Smoke/rooms.js (hunk 19 above)
Comment (BrayanRosas) [inline tests/Smoke/rooms.js]: this already refactor

**Comment by AlejandroJeanCarlo** [comment]
Something in your code is causing conflicts, please review again

File: config/room.json [hunk 20]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (AlejandroJeanCarlo) [inline config/room.json]: re factor please

**Pull request merged**